src/main/java/org/teleight/teleightbots/scheduler/Scheduler.java

```diff
@@ -14,7 +14,7 @@
  *
  * @see SchedulerImpl
  */
-public interface Scheduler extends Closeable {
+public sealed interface Scheduler extends Closeable permits SchedulerImpl {
```

Comment:
I think the scheduler is one of those components that should be extensible by developers